.github/flaky_test_detection.py

```diff
@@ -0,0 +1,114 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import requests
+from github import Github
+from github import Auth
+
+
+GIT_ORG = "apache"
+GIT_REPO = "beam"
+GRAFANA_URL = "http://metrics.beam.apaache.org"
+#
+ALERT_NAME = "flaky_test"
+READ_ONLY = os.environ.get("READ_ONLY", "false")
+
+
+class Alert:
+    def __init__(self, workflow_id, workflow_url, workflow_name, workflow_filename):
+        self.workflow_id = workflow_id
+        self.workflow_url = workflow_url
+        self.workflow_name = workflow_name
+        self.workflow_file_name = workflow_filename
+
+
+def extract_workflow_id_from_issue_label(issues):
+    label_ids = []
+    for issue in issues:
+        for label in issue.get_labels():
+            match = re.search(r"workflow_id:\s*(\d+)", str(label.name))
+            if match:
+                label_id = match.group(1)
+                label_ids.append(label_id)
+
+    return label_ids
+
+
+def create_github_issue(repo, alert):
+    failing_runs_url = f"https://github.com/{GIT_ORG}/beam/actions/{alert.workflow_file_name}?query=is%3Afailure+branch%3Amaster"
+    title = f"The {alert.workflow_name} job is flaky"
+    body = f"The {alert.workflow_name } is constantly failing.\nPlease visit {failing_runs_url} to see the logs."
```

Comment:
Can we propogate through/add the threshold value here? Something like `f"The {alert.workflow_name } job is failing over XX% of the time.\nPlease visit {failing_runs_url} to see the logs."